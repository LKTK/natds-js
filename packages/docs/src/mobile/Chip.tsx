import * as React from 'react';
import { Platform } from 'react-native';
import { Chip } from '@naturacosmeticos/natds-rn';
import withJest from '../decorators/jest';
import { text } from '@storybook/addon-knobs';

export default {
  title: 'Mobile|Chip',
  component: Chip,
  decorators: [withJest()],
  parameters: { jestImportPath: 'mobile', jest: ['Chip'] }
};

export const Default = () => (
  <>
    {Platform.OS === 'web' ? (
      <style type="text/css">{`
    @font-face {
      font-family: 'MaterialIcons';
      src: url('${require('react-native-vector-icons/Fonts/MaterialIcons.ttf')}') format('truetype');
    }
  `}</style>
    ) : null}
    <Chip
      children={text('children', 'Native Chip')}
      onClose={() => {}}
      onPress={() => {}}
    />
  </>
);
