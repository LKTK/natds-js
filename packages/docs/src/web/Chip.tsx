import * as React from 'react';
import { Chip } from '@natds/web';
import withJest from '../decorators/jest';

export default {
  title: 'Web|Chip',
  component: Chip,
  decorators: [withJest()],
  parameters: {
    jestImportPath: 'web',
    jest: ['Chip']
  }
};

export const Default = () => (
  <Chip
    label="Some Label"
    onDelete={() => { }}
    clickable={true}
  />
);
