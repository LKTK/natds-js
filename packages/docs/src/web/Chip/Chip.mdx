import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import { Chip } from '@naturacosmeticos/natds-web';

# Chip

I can embed a story (but not define one, since this file should not contain a `Meta`):
