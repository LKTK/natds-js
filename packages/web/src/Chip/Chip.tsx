import React, { FunctionComponent, useState } from 'react';

export interface IProps {
  labelOn: string;
  labelOff: string;
};

const Chip: FunctionComponent<IProps> = (props: IProps) => {
  const [isOn, toggleCheck] = useState(false);

  return (
    <label>
      <input
        type="checkbox"
        checked={isOn}
        onChange={() => toggleCheck(!isOn)}
      />
      {isOn ? props.labelOn : props.labelOff}
    </label>
  );
};

export default Chip;
