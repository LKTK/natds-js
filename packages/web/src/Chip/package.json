{
  "name": "Chip",
  "main": "./Chip.tsx",
  "private": true
}
