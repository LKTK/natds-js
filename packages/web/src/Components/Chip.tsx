import React, { FunctionComponent } from 'react';

import MaterialChip from '@material-ui/core/Chip';
import { withTheme } from '@material-ui/styles';

export interface IProps {
  avatar?: any;
  classes?: object;
  clickable?: boolean;
  component?: any;
  deleteIcon?: any;
  disabled?: boolean;
  icon?: any;
  label: string;
  onDelete?: () => void;
}

const Chip: FunctionComponent<IProps> = (props: IProps) => {
  return (
    <MaterialChip {...props} />
  );
};

export default withTheme(Chip);
