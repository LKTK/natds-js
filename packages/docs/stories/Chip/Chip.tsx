import React from 'react';
import { Chip } from '@natds/web';

export default { title: 'Chip' };

export const withText = () => (
  <Chip labelOn="On" labelOff="off" />
);
