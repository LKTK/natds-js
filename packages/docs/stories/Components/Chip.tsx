import * as React from 'react';
import { Chip } from '@natds/web';
import Button from '@natds/mobile/dist/Components/Button';

export default {
  title: 'Components|Chip',
  component: Chip
};

export const WithDeleteAction = () => (
  <Chip label="Some Label" onDelete={() => {}} />
);

export const Native = () => <Button />;
