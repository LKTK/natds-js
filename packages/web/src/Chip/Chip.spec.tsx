
import React from 'react';
import { mount } from 'enzyme';
import { expect } from 'chai';
import { assert, spy } from 'sinon'

import { Chip } from './Chip';

describe('Chip component', () => {
  test('Chip calls onDelete', () => {
    const onDelete = spy()
    const label = 'A Chip Component'
    const chip = mount(<Chip label={label} onDelete={onDelete} />);

    expect(chip.find('.MuiChip-label')).to.have.text(label);

    chip.find('.MuiChip-deleteIcon').first().simulate('click');

    assert.calledOnce(onDelete);

    expect(chip).to.matchSnapshot();
  })
})
