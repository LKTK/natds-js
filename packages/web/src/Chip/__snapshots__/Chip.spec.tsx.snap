// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component Chip changes the text after click 1`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  On
</label>
`;
